bin/lambda-cfn-rules.js

```diff
@@ -3,49 +3,206 @@
+
+
+
+if (process.env.NODE_ENV == 'test') {
+  var files = fs.readdirSync(path.join(process.cwd(), 'test/cloudformation'));
+} else {
+  var files = fs.readdirSync(path.join(process.cwd(), 'cloudformation'));
+}
 
 for (var i = 0; i < files.length; i++) {
   var file = files[i];
   if (path.extname(file) == '.js' && file.indexOf('.template.') > -1) {
-    templateFile = path.join(process.cwd(), 'cloudformation', file);
+    if (process.env.NODE_ENV == 'test') {
+      templateFile = path.join(process.cwd(), 'test/cloudformation/lambda-rules.template.js');
+    } else {
+      templateFile = path.join(process.cwd(), 'cloudformation', file);
+    }
     break;
   }
 }
 
 var template = require(templateFile);
+
 for (var r in template.Resources) {
   if (template.Resources[r].Type == 'AWS::Lambda::Function' &&
       template.Resources[r].Metadata &&
       template.Resources[r].Metadata.sourcePath) {
     var sourcePath = path.join(process.cwd(), template.Resources[r].Metadata.sourcePath);
     var rule = require(sourcePath);
+    rule.config.stackRuleName = argv.stack + '-' + rule.config.name;
     rules.push(rule);
   }
 }
 
-var stackName = process.argv[2];
+var stackName = argv.stack;
 var region = process.env.AWS_DEFAULT_REGION || 'us-east-1';
-var q = queue(1);
+var q = d3.queue(1);
 var cfn = new AWS.CloudFormation({region: region});
 var cwe = new AWS.CloudWatchEvents({region: region});
 var lambda = new AWS.Lambda({region: region});
 
-q.defer(getStackResources);
-q.defer(createEventRules);
-q.awaitAll(function(err) {
-  if (err) throw err;
-  else console.log('CloudWatch Event Rules created');
-});
+if (argv._ == 'create') {
+  q.defer(getStackResources);
+  q.defer(createEventRules);
+  q.awaitAll(function(err) {
+    if (err) throw err;
+    else console.log('CloudWatch Event Rules creation complete');
+  });
+}
+
+if (argv._ == 'delete') {
+  q.defer(getMatchedRules);
+  q.defer(listRules);
+  q.defer(deleteRules);
+  q.awaitAll(function(err) {
+    if (err) throw err;
+    else console.log('CloudWatch Event Rules deletion complete');
+  });
+}
+
+if (argv._ == 'list') {
+  q.defer(getMatchedRules);
+  q.defer(listRules);
+  q.awaitAll(function(err) {
+    if (err) throw err;
+  });
+}
+
+function deleteRules(callback) {
+  inquirer.prompt([
+    {
+      type: 'confirm',
+      message: 'Please confirm the deletion of the CloudWatch rules',
+      name: 'delete',
+      default: false
+    }
+  ]).then(function(answers) {
+    if (answers.delete == true) {
+      deleteRuleset(matchedRules, function(err){
+        if (err) return callback(err);
+        return callback();
+      });
+    } else {
+      callback();
+    }
+  });
+}
+
+function listRules(callback) {
+  if (matchedRules.length > 0) {
+    matchedRules.forEach(function(rule) {
+      console.log(rule.config.stackRuleName);
+    });
+    callback();
+  } else {
+    console.log('No matching rules found');
+    return new Error();
```

Comment:
Any reason not to do here: return callback(new Error('No matching rules found')); ?  I see you throw the err back where listRules gets called, so it'll get printed there.